File: swagger/models/appointment.py
from .base_model import BaseModel


class Appointment(BaseModel):
	"""
	Contains information about an appointment.

	Attributes:

		gender_preference
		The preferred gender of the appointment provider.

		duration
		The duration of the appointment.

		provider_id
		If a user has Complementary and Alternative Medicine features enabled,
		this property indicates the provider assigned to the appointment.

		id
		The unique ID of the appointment.

		status
		The status of this appointment.

		start_date_time
		The date and time the appointment is to start.

		end_date_time
		The date and time the appointment is to end.

		notes
		Any notes associated with the appointment.

		staff_requested
		When `true`, indicates that the staff member was requested
		specifically by the client.

		program_id
		The ID of the program to which this appointment belongs.

		session_type_id
		The ID of the session type of this appointment.

		location_id
		The ID of the location where this appointment is to take place.

		staff_id
		The ID of the staff member providing the service for this appointment.

		client_id
		The RSSID of the client who is booked for this appointment.

		first_appointment
		When `true`, indicates that this is the client’s first appointment
		at this site.

		client_service_id
		The ID of the pass on the client’s account that is to pay for this
		appointment.

		resources
		The resources this appointment is to use.
	"""

	input_map = {
		'GenderPreference': 'gender_preference',
		'Duration': 'duration',
		'ProviderId': 'provider_id',
		'Id': 'id',
		'Status': 'status',
		'StartDateTime': 'start_date_time',
		'EndDateTime': 'end_date_time',
		'Notes': 'notes',
		'StaffRequested': 'staff_requested',
		'ProgramId': 'program_id',
		'SessionTypeId': 'session_type_id',
		'LocationId': 'location_id',
		'StaffId': 'staff_id',
		'ClientId': 'client_id',
		'FirstAppointment': 'first_appointment',
		'ClientServiceId': 'client_service_id',
		'Resources': 'resources',
		}

	output_map = {
		'gender_preference': 'GenderPreference',
		'duration': 'Duration',
		'provider_id': 'ProviderId',
		'id': 'Id',
		'status': 'Status',
		'start_date_time': 'StartDateTime',
		'end_date_time': 'EndDateTime',
		'notes': 'Notes',
		'staff_requested': 'StaffRequested',
		'program_id': 'ProgramId',
		'session_type_id': 'SessionTypeId',
		'location_id': 'LocationId',
		'staff_id': 'StaffId',
		'client_id': 'ClientId',
		'first_appointment': 'FirstAppointment',
		'client_service_id': 'ClientServiceId',
		'resources': 'Resources',
		}

	gender_preference = None
	duration = None
	provider_id = None
	id = None
	status = None
	start_date_time = None
	end_date_time = None
	notes = None
	staff_requested = None
	program_id = None
	session_type_id = None
	location_id = None
	staff_id = None
	client_id = None
	first_appointment = None
	client_service_id = None
	resources = None

File: swagger/Models/Appointment.php
<?php

namespace App\MindBody\Models;

/**
 * Contains information about an appointment.
 */
class Appointment extends BaseModel
{
	/**
	 * The preferred gender of the appointment provider.
	 * 
	 * @var string 
	 */
	public $genderPreference;
	/**
	 * The duration of the appointment.
	 * 
	 * @var integer format:int32
	 */
	public $duration;
	/**
	 * If a user has Complementary and Alternative Medicine features enabled,
	 * this property indicates the provider assigned to the appointment.
	 * 
	 * @var string 
	 */
	public $providerId;
	/**
	 * The unique ID of the appointment.
	 * 
	 * @var integer format:int64
	 */
	public $id;
	/**
	 * The status of this appointment.
	 * 
	 * @var string 
	 */
	public $status;
	/**
	 * The date and time the appointment is to start.
	 * 
	 * @var string format:date-time
	 */
	public $startDateTime;
	/**
	 * The date and time the appointment is to end.
	 * 
	 * @var string format:date-time
	 */
	public $endDateTime;
	/**
	 * Any notes associated with the appointment.
	 * 
	 * @var string 
	 */
	public $notes;
	/**
	 * When `true`, indicates that the staff member was requested
	 * specifically by the client.
	 * 
	 * @var boolean 
	 */
	public $staffRequested;
	/**
	 * The ID of the program to which this appointment belongs.
	 * 
	 * @var integer format:int32
	 */
	public $programId;
	/**
	 * The ID of the session type of this appointment.
	 * 
	 * @var integer format:int32
	 */
	public $sessionTypeId;
	/**
	 * The ID of the location where this appointment is to take place.
	 * 
	 * @var integer format:int32
	 */
	public $locationId;
	/**
	 * The ID of the staff member providing the service for this appointment.
	 * 
	 * @var integer format:int64
	 */
	public $staffId;
	/**
	 * The RSSID of the client who is booked for this appointment.
	 * 
	 * @var string 
	 */
	public $clientId;
	/**
	 * When `true`, indicates that this is the client’s first appointment
	 * at this site.
	 * 
	 * @var boolean 
	 */
	public $firstAppointment;
	/**
	 * The ID of the pass on the client’s account that is to pay for this
	 * appointment.
	 * 
	 * @var integer format:int64
	 */
	public $clientServiceId;
	/**
	 * The resources this appointment is to use.
	 * 
	 * @var Resource[] 
	 */
	public $resources;

	protected function getInputMap()
	{
		return [
			'GenderPreference' => ['genderPreference', 'string', null],
			'Duration' => ['duration', 'integer', 'int32'],
			'ProviderId' => ['providerId', 'string', null],
			'Id' => ['id', 'integer', 'int64'],
			'Status' => ['status', 'string', null],
			'StartDateTime' => ['startDateTime', 'string', 'date-time'],
			'EndDateTime' => ['endDateTime', 'string', 'date-time'],
			'Notes' => ['notes', 'string', null],
			'StaffRequested' => ['staffRequested', 'boolean', null],
			'ProgramId' => ['programId', 'integer', 'int32'],
			'SessionTypeId' => ['sessionTypeId', 'integer', 'int32'],
			'LocationId' => ['locationId', 'integer', 'int32'],
			'StaffId' => ['staffId', 'integer', 'int64'],
			'ClientId' => ['clientId', 'string', null],
			'FirstAppointment' => ['firstAppointment', 'boolean', null],
			'ClientServiceId' => ['clientServiceId', 'integer', 'int64'],
			'Resources' => ['resources', 'array', Resource::class],
			];
	}

	protected function getOutputMap()
	{
		return [
			'genderPreference' => ['GenderPreference', 'string', null],
			'duration' => ['Duration', 'integer', 'int32'],
			'providerId' => ['ProviderId', 'string', null],
			'id' => ['Id', 'integer', 'int64'],
			'status' => ['Status', 'string', null],
			'startDateTime' => ['StartDateTime', 'string', 'date-time'],
			'endDateTime' => ['EndDateTime', 'string', 'date-time'],
			'notes' => ['Notes', 'string', null],
			'staffRequested' => ['StaffRequested', 'boolean', null],
			'programId' => ['ProgramId', 'integer', 'int32'],
			'sessionTypeId' => ['SessionTypeId', 'integer', 'int32'],
			'locationId' => ['LocationId', 'integer', 'int32'],
			'staffId' => ['StaffId', 'integer', 'int64'],
			'clientId' => ['ClientId', 'string', null],
			'firstAppointment' => ['FirstAppointment', 'boolean', null],
			'clientServiceId' => ['ClientServiceId', 'integer', 'int64'],
			'resources' => ['Resources', 'array', Resource::class],
			];
	}
}
